nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -657,11 +657,23 @@ Costmap2DROS::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameter
       }
     } else if (type == ParameterType::PARAMETER_INTEGER) {
       if (name == "width") {
-        resize_map = true;
-        map_width_meters_ = parameter.as_int();
+        if (parameter.as_int() > 0) {
+          resize_map = true;
+          map_width_meters_ = parameter.as_int();
+        } else {
+          RCLCPP_ERROR(
+            get_logger(), "You try to set width of map to be negative or zero,"
+            " this isn't allowed, please give a positive value.");
```

Comment:
This should also set `result.successful = false;` and return to indicate the update failed